### ISSUE TITLE
Vscode coloring is broken for large files

### ISSUE THREAD

**Issue by rchiodo** [Microsoft/vscode#11420]
This issue was moved. Not sure how to reopen:
https://github.com/Microsoft/vscode/issues/10901

This should be a vscode issue, not an extension issue. The extension doesn't control colorization.

**Comment by aeschli** [comment]
@rchiodo Can you attach the code example where you see the problem?

**Comment by angrah** [comment]
Here is an example test.json:

```
{
	"a": {
		"b": {
			"c": {
				"d": {
					"e": {
						"f": {
							"g": {
								"h": {
									"i": {
										"j": {
											"k": {
												"l": {
													"m": {
														"n": {
															"o": {
																"p": "test"
                                                            }
                                                        }
                                                    }
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}

```
Line "p" and anything below is not colored.

**Comment by hashhar** [comment]
A related issue is that searching for a token doesn't highlight more than 500 entries at a time.

For example I have a latex style file which has about 600 lines and almost all of the lines contain the text `faicon`. Searching for it only highlights the first 500 matches. And also there is not configurable way to increase that number.

**Comment by mjbvz** [comment]
@angrah Your example is fixed in vscode 1.7+:

<img width="687" alt="screen shot 2016-11-18 at 4 19 49 pm" src="https://cloud.githubusercontent.com/assets/12821956/20451111/c5b5de4c-adaa-11e6-8311-21a3dbd85c63.png">


@rchiodo Extensions provide the grammars that control language colorization. Can you please try out your example in VSCode 1.7+ to see if you still see these errors. If you do, please upload an example file that demonstrates this problem so that we can investigate further.

Thanks.

**Comment by hashhar** [comment]
@mjbvz My issue got fixed in v1.7. I don't know if it was an extension update or VSCode update. Thanks.
